core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -346,7 +346,7 @@ case class Stop(id: stop.Stop, sourceInfo: SourceInfo, clock: Arg, ret: Int) ext
   "This API should never have been public, for Module port reflection, use DataMirror.modulePorts",
   "Chisel 3.5"
 )
-case class Port(id: Data, dir: SpecifiedDirection, sourceInfo: SourceInfo)
+case class Port(id: IsBindable, dir: SpecifiedDirection, sourceInfo: SourceInfo)
```

Comment:
This is the actual motivation for this change: I want to be able to have Ports that aren't necessarily Data (i.e., could be Property).

Reply:
As part of merging this, we really need to make all of these case classes package private (which we can do now, see the deprecation), but obviously not part of this PR.

Reply:
Should the deprecated classes be made private before this change? Or it's just something we should prioritize?

Reply:
It's just something we should prioritize, doesn't block this change (although any code ignoring the deprecation warning might break)